flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/reader/splitreader/SplitsRemoval.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.base.source.reader.splitreader;
+
+import org.apache.flink.annotation.Internal;
+
+import java.util.List;
+
+/**
+ * A change to remove splits.
+ *
+ * <p>This type of {@link SplitsChange} is only used when the {@link
+ * org.apache.flink.connector.base.source.reader.RecordEvaluator} reports the end of the stream. The
+ * split removal between the enumerator and reader still exists some problems.
```

Comment:
It would be really helpful to me if you could elaborate on what these problems are. Are they some technical issues that are hard to resolve, or are they just left to be done?

Reply:
I remember the community have discussed about the split deletion issue. And there may be some problems about the state if we introduce the split deletion. But I can not find that mail. There is no plan on my side for that issue now.

Reply:
Thanks, I'll look into it.